pkg/ui/terminal.go

```diff
@@ -420,3 +425,44 @@ func (u *TerminalUI) handleMessage(msg *api.Message) {
 func (u *TerminalUI) ClearScreen() {
 	fmt.Print("\033[H\033[2J")
 }
+
+func formatToolCallResponse(payload any) string {
+	if payload == nil {
+		return ""
+	}
+
+	// If payload is a JSON string try to unmarshal first
+	if s, ok := payload.(string); ok {
+		var m map[string]any
+		if err := json.Unmarshal([]byte(s), &m); err == nil {
+			return formatToolCallResponse(m)
+		}
+		return truncateString(s, 1000)
+	}
+
+	if m, ok := payload.(map[string]any); ok {
+		if stdout, ok := m["stdout"]; ok {
+			if outStr, ok := stdout.(string); ok {
+				return truncateString(outStr, 1000)
+			}
+		}
+		b, err := json.MarshalIndent(m, "", "  ")
+		if err == nil {
+			return truncateString(string(b), 1000)
+		}
+		return fmt.Sprint(m)
+	}
```

Comment:
curious why we have to check for string and map ouput, we can constraint the output to always follow `map` format.